File: Min Stack - Leetcode 155/Min Stack - Leetcode 155.cpp
#include <stack>
using namespace std;

class MinStack {
public:
    MinStack() {}

    void push(int val) {
        stk.push(val);
        if (minStk.empty() || val <= minStk.top()) {
            minStk.push(val);
        } else {
            minStk.push(minStk.top());
        }
    }

    void pop() {
        stk.pop();
        minStk.pop();
    }

    int top() {
        return stk.top();
    }

    int getMin() {
        return minStk.top();
    }

private:
    stack<int> stk;
    stack<int> minStk;
};

File: Min Stack - Leetcode 155/Min Stack - Leetcode 155.js
class MinStack {
    constructor() {
        this.stk = [];
        this.minStk = [];
    }

    push(val) {
        this.stk.push(val);
        if (this.minStk.length === 0 || val <= this.minStk[this.minStk.length - 1]) {
            this.minStk.push(val);
        } else {
            this.minStk.push(this.minStk[this.minStk.length - 1]);
        }
    }

    pop() {
        this.stk.pop();
        this.minStk.pop();
    }

    top() {
        return this.stk[this.stk.length - 1];
    }

    getMin() {
        return this.minStk[this.minStk.length - 1];
    }
}
